march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -117,7 +117,11 @@ void MarchHardwareInterface::init()
         errorStream << "Joint " << joint_names_[i].c_str() << " has no net number";
         throw std::runtime_error(errorStream.str());
       }
-      marchRobot.getPowerDistributionBoard()->getHighVoltage().setNetOnOff(false, netNumber);
+      while(!marchRobot.getPowerDistributionBoard()->getHighVoltage().getNetOperational(netNumber)){
+        marchRobot.getPowerDistributionBoard()->getHighVoltage().setNetOnOff(true, netNumber);
+        usleep(100000);
```

Comment:
Add an explanation here

Reply:
Why do you think this code is not readable/understandable?

Reply:
Ah nevermind, misread the sleep location

Reply:
Add a timeout here though?

Reply:
Not needed